### PR TITLE
Issue 98 - create test report in JUnit format

### PR DESCRIPTION
**Pull request by lbulanti-ms**
In this PR I add a new cli option to specify a location where to generate the test evidence JUnit report file. 

As part of this PR I add prefixes and suffixes to the spectral ruleset for instantiation and pattern to make clear where the issue is happening

### CODE REVIEW

File: cli/src/commands/validate/validate.spec.ts [hunk 1]
```diff
@@ -203,6 +212,36 @@ describe('validate', () => {
...
+
+        const apiGateway = readFileSync(path.resolve(__dirname, '../../../test_fixtures/api-gateway.json'), 'utf8');
+        fetchMock.mock('http://exist/api-gateway.json', apiGateway);
+
+        const apiGatewayInstantiation = readFileSync(path.resolve(__dirname, '../../../test_fixtures/api-gateway-implementation.json'), 'utf8');
+        fetchMock.mock('https://exist/api-gateway-implementation.json', apiGatewayInstantiation);
+
+        const tmpDir = mkdtempSync(path.join(os.tmpdir()));
+        const testReportLocation = tmpDir + '/test-report.xml';
+        console.log(testReportLocation);
```
Comment (lbulanti-ms) [inline cli/src/commands/validate/validate.spec.ts:232]: remove this line

**Comment by lbulanti-ms** [comment]
In this PR I added to the spectral rules a prefix or a suffix to make sure that is easy to understand if the issues happened in the pattern or in the instantiation. This change was required because lots of the rules had the same name in the both rulesets.

File: cli/src/commands/validate/junit-report/junit.report.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import junitReportBuilder, { TestSuite } from 'junit-report-builder';
+import { ValidationOutput } from '../validation.output';
+
+export default function createJUnitReport(
+    jsonSchemaValidationOutput: ValidationOutput[], 
+    spectralValidationOutput: ValidationOutput[], 
+    spectralRules: string[],
```
Comment (grahampacker-ms) [inline cli/src/commands/validate/junit-report/junit.report.ts:7]: rename to `spectralRuleNames`

File: cli/src/commands/validate/junit-report/junit.report.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    builder.writeTo(outputLocation);
+}
+
+function createTestSuite(builder, testSuiteName: string){
+    return builder
+        .testSuite()
+        .name(testSuiteName);
+}
+
+function createTestCase(testSuite: TestSuite, testName: string){
```
Comment (grahampacker-ms) [inline cli/src/commands/validate/junit-report/junit.report.ts:49]: rename `createSucceedingTestCase`

**Review by willosborne — APPROVED** [review]

**Review by aidanm3341 — APPROVED** [review]

**Pull request merged**